angularSTEP/src/app/upload-recipe/upload-recipe.component.ts

```diff
@@ -160,4 +196,30 @@ export class UploadRecipeComponent{
       });
     this.router.navigate(['/confirm-upload']);
   }
+
+  //image functions
+
+  onFileSubmit() {
+    this.fAuth.currentUser.then(user => {
+      if (user) {
+        //this.addImage(user.uid);
+      } else {
+        this._ngZone.run(() => {
+          this.router.navigate(['/home']);
+        });
+      }
+    });
+  }
+}
+
+function requiredFileType(control: FormControl) {
+  const imageTypes: Array<string> = ['png', 'jpg', 'jpeg', 'gif'];
+  const file = control.value;
+  if (file && file.split('.')[1]) {
```

Comment:
I think this explicitly disallows a file named my.cool.picture.jpg. Probably fine, as long as it's intentional.

Reply:
Yes I had realized that. I think that is actually good because I am not sure the naming schemas that Firebase Storage allows. This keeps it uniform while also maintaining a little validation.